### ISSUE TITLE
Save and refresh/ revert don't necessary clear dirty attribute at point of changed signal.

### ISSUE THREAD

**Issue by 0Hughman0** [0Hughman0/jupyter_cassini#27]
Version 0.2.0a5 and below.

Pressing the save or fetch from disk buttons in the tier browser don't necessarily result in the dirty (*) indicator being removed.

This is because the save and revert methods of `TierModel` use `metaFile.save` and `metaFile.revert` behind the scenes.

Both these methods [call the private method `Context._updateContentsModel`](https://github.com/jupyterlab/jupyterlab/blob/66351c7d2f350756ca0436abc66e6c9848e0c6af/packages/docregistry/src/context.ts#L515), which is what emits the `changed` signal the model listens to.

The dirty attribute of the `Context` is only set _after_ `_updateContentsModel` is called, therefore the signal is emitted before the dirty state is updated.

1. Save ... hmm doesn't seem to actually be the case, so I'm confused!
2. [Revert](https://github.com/jupyterlab/jupyterlab/blob/de28aca9ca8a60b25d28efdf359bb2c69d58c634/packages/docregistry/src/context.ts#L660)

This means when `TierModel.changed` is emitted, the `dirty` attribute of `metaFile` might not have been updated yet.

**Comment by 0Hughman0** [comment]
Seems to be only a bug in version 4.0.4...?

**Comment by 0Hughman0** [comment]
Likely caused by this change:

https://github.com/jupyterlab/jupyterlab/pull/14771/files#diff-bd91309321cead1ad13aa66f461c81110113c117fababf12ea04e346b0b5afb5L491

Which is probably fair enough, saving the model doesn't change its contents.

Issue with revert still the case though.

In our case, saving the model _really does change it_ (it changes dirty!).

**Comment by 0Hughman0** [comment]
Currently using `metaFile.model.sharedModel.changed` to propagate any changes forward. 

As `dirty` is also part of the state of the model, we should also listen to changes to this.

Best solution I think is to also bind changes to `metaFile.model.stateChanged`, which includes changes to dirty!

```
metaFile.model.stateChanged.connect((sender, change) => {
          if (change.name == 'dirty') {
            this._changed.emit()
          }
        })
```

**Comment by 0Hughman0** [comment]
Fixed in #29
